pandas/indexes/numeric.py

```diff
@@ -155,17 +164,22 @@ def equals(self, other):
         if self.is_(other):
             return True
 
-        try:
-            return com.array_equivalent(com._values_from_object(self),
-                                        com._values_from_object(other))
-        except TypeError:
-            # e.g. fails in numpy 1.6 with DatetimeIndex #1681
-            return False
+        return com.array_equivalent(com._values_from_object(self),
+                                    com._values_from_object(other))
 
     def _wrap_joined_index(self, joined, other):
         name = self.name if self.name == other.name else None
         return Int64Index(joined, name=name)
 
+    @classmethod
+    def _check_casting(cls, data, subarr):
```

Comment:
I prefer ``_assert_xxx`` for validation method, like existing ``_assert_can_do_op`` and ``_assert_take_fillable``